cmds/report.c

```diff
@@ -43,15 +43,19 @@ static void find_insert_node(struct rb_root *root, struct uftrace_task_reader *t
 			     uint64_t timestamp, uint64_t addr, bool needs_srcline)
 {
 	struct uftrace_symbol *sym;
+	struct uftrace_symbol *origin_sym;
 	char *symname;
 	struct uftrace_dbg_loc *loc = NULL;
 
 	sym = task_find_sym_addr(&task->h->sessions, task, timestamp, addr);
 	if (needs_srcline)
 		loc = task_find_loc_addr(&task->h->sessions, task, timestamp, addr);
 
+	origin_sym = task->func;
+	task->func = sym;
 	symname = symbol_getname(sym, addr);
 	insert_node(root, task, symname, loc);
+	task->func = origin_sym;
```

Comment:
I don't understand why this is necessary.  Could you please add comments?

Reply:
That's because I had no idea of which cause side effect.
But now, I think that is unnecessary too.